api/coreservice.go

```diff
@@ -404,8 +400,7 @@ func (core *coreService) PendingNonce(addr address.Address) (uint64, error) {
 }
 
 func (core *coreService) validateChainID(chainID uint32) error {
-	if core.cfg.Genesis.Blockchain.IsMidway(core.bc.TipHeight()) &&
-		chainID != core.bc.ChainID() && chainID != 0 {
+	if ge := core.bc.Genesis(); ge.IsMidway(core.bc.TipHeight()) && chainID != core.bc.ChainID() && chainID != 0 {
```

Comment:
nit: no need to define `ge`? since it is not used later

Reply:
it won't work.